base/Cargo.toml

```diff
@@ -0,0 +1,15 @@
+[package]
+name = "base"
```

Comment:
What's base? This needs a better name and some documentation in a readme or somewhere to explain what it is.

Reply:
Yeah... it's a bad name. `base` provides the `RecordStream` and some debug functions (to debug records) that are used in both `simple_https_client` and `simple_https_server`. Maybe `record`?

Reply:
yeah something like `record` or `record_layer` might be good.
If there should be more in here later, we could also get a more generic `?_lib` name, like `app_lib` or whatever.